### ISSUE TITLE
Rename WorldObjectShader.type to WorldObjectShader.kind

### ISSUE THREAD

**Issue by FirefoxMetzger** [pygfx/pygfx#409]
I am looking into the way shaders are written in pygfx. According to the docs, this is done by subclassing `WorldObjectShader` which has to define a class-level variable called `type`. Would it be useful to rename this variable to `kind` in order to avoid aliasing the built-in `type` function?

**Comment by almarklein** [comment]
Yeah, that makes sense. The name does not matter so much, and avoiding a builtin is probably a good idea 👍

**Comment by kushalkolar** [comment]
Do you want to do it for `Event.type` as well? (sorry if you prefer this in a separate issue)

https://github.com/pygfx/pygfx/blob/main/pygfx/objects/_events.py#L66

**Comment by FirefoxMetzger** [comment]
I think that would make sense.

**Comment by Korijn** [comment]
I'm not 100% clear on how it shadows the built-in function. It's not in globals or locals, right? You can only access it as an attribute.
